src/main/kotlin/no/nav/oppgavelytter/oppgave/kafka/OppgaveConsumer.kt

```diff
@@ -0,0 +1,56 @@
+package no.nav.oppgavelytter.oppgave.kafka
+
+import no.nav.oppgavelytter.oppgave.OppgaveService
+import no.nav.sykdig.applog
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import java.time.Duration
+
+class OppgaveConsumer(
+    private val oppgaveTopic: String,
+    private val kafkaConsumer: KafkaConsumer<String, OppgaveKafkaAivenRecord>,
+    private val oppgaveService: OppgaveService,
+    private val applicationState: no.nav.sykdig.ApplicationState,
+) {
+    val logger = applog()
+
+    suspend fun startConsumer() {
+        while (applicationState.ready) {
+            try {
+                kafkaConsumer.subscribe(listOf(oppgaveTopic))
+                consume()
+            } catch (ex: Exception) {
+                logger.error("error running oppgave-consumer", ex)
+                kafkaConsumer.unsubscribe()
+                logger.info(
+                    "Unsubscribed from topic $oppgaveTopic and waiting for 10 seconds before trying again",
+                )
+            }
+        }
+    }
```

Comment:
det er kanskje mer noe native støtte for kafka i Spring Boot vi kunne benyttet oss av? :thinking:

Reply:
Jess, det kan jeg absolutt sjekke

Reply:
> det er kanskje mer noe native støtte for kafka i Spring Boot vi kunne benyttet oss av? 🤔

Har lagt på KafkaListener og skrevet om koden